src/main/java/org/folio/service/group/GroupService.java

```diff
@@ -1,100 +1,57 @@
 package org.folio.service.group;
 
-import static org.folio.rest.jaxrs.resource.FinanceStorageGroups.PostFinanceStorageGroupsResponse.headersFor201;
-import static org.folio.rest.util.ResponseUtils.buildErrorResponse;
-
-import java.util.UUID;
-
-import javax.ws.rs.core.Response;
+import java.util.List;
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.folio.builders.error.NameCodeConstraintErrorBuilder;
+import org.folio.dao.group.GroupDAO;
+import org.folio.rest.core.model.RequestContext;
+import org.folio.rest.jaxrs.model.Budget;
 import org.folio.rest.jaxrs.model.Group;
-import org.folio.rest.jaxrs.resource.FinanceStorageGroups;
-import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.jaxrs.model.GroupFundFiscalYear;
+import org.folio.rest.persist.DBClient;
+import org.folio.rest.persist.DBClientFactory;
+import org.folio.rest.persist.DBConn;
 
-import io.vertx.core.AsyncResult;
-import io.vertx.core.Context;
 import io.vertx.core.Future;
-import io.vertx.core.Handler;
-import io.vertx.core.Promise;
-import io.vertx.core.Vertx;
-import io.vertx.core.json.JsonObject;
-import org.folio.rest.exception.HttpException;
 
 public class GroupService {
 
   private static final Logger logger = LogManager.getLogger(GroupService.class);
 
-  private static final String GROUPS_TABLE = "groups";
+  private final DBClientFactory dbClientFactory;
+  private final GroupDAO groupDAO;
 
-  private PostgresClient pgClient;
-  private final NameCodeConstraintErrorBuilder nameCodeConstraintErrorBuilder;
-
-  public GroupService(Vertx vertx, String tenantId) {
-    this.pgClient = PostgresClient.getInstance(vertx, tenantId);
-    this.nameCodeConstraintErrorBuilder = new NameCodeConstraintErrorBuilder();
+  public GroupService(DBClientFactory dbClientFactory, GroupDAO groupDAO) {
+    this.dbClientFactory = dbClientFactory;
+    this.groupDAO = groupDAO;
   }
 
-  public void createGroup(Group entity, Context vertxContext, Handler<AsyncResult<Response>> asyncResultHandler) {
+  public Future<Group> createGroup(Group entity, RequestContext requestContext) {
     logger.debug("createGroup:: Trying to create group");
-    vertxContext.runOnContext(v -> createGroup(entity)
-      .onSuccess(group -> {
-        logger.info("createGroup:: Group with id {} created", entity.getId());
-        asyncResultHandler.handle(Future.succeededFuture(
-            FinanceStorageGroups.PostFinanceStorageGroupsResponse.respond201WithApplicationJson(group, headersFor201())));
-      })
-      .onFailure(throwable -> {
-        logger.error("createGroup:: Group creation with id {} failed", entity.getId(), throwable);
-        asyncResultHandler.handle(buildErrorResponse(throwable));
-      }));
+    DBClient client = dbClientFactory.getDbClient(requestContext);
+    return client.withTrans(conn -> groupDAO.createGroup(entity, conn));
```

Comment:
If it is single operation then you can use client.withConn() to not create new transaction

Reply:
The problem is that from the service point of view, it is not known if a DAO layer will use the connexion once or more. Also it could change in the future, and the service should keep working if the DAO is modified. And I am trying to keep the calls consistent from one service to another. Is there much of an overhead in using a transaction ?